### PR TITLE
[IAC-15] - Create Ably API Key Resource

### PR DESCRIPTION
**Pull request by graham-russell**
Includes resource declaration + tests. Also includes terraform templates in playground and resource examples.
NOTE: We had previously planned to create two distinct resources - ably_api_key and ably_api_key_capability.
But because the Control API won't allow the creation of an Ably Key without at least one capability being specified, this made it tricky.
In the case where a key & key_capability resource are both declared:
- You would need to provide at least one capability inline in the ably_api_key resource
- You then create ably_api_key_capability resource which updates the Key with new capabilities.
Unfortunately, this results in a situation where the capability created in ably_api_key is overwritten and you land up in a case where all capabilities are not created.

Test Output:
```
~/Documents/src/terraform-provider-ably (feature/IAC-15-Key-Resource ✗) make testacc
TF_ACC=1 go test $(go list ./... | grep -v 'vendor') -v  -timeout 120m
?       terraform-provider-ably [no test files]
=== RUN   TestAccAblyApp
--- PASS: TestAccAblyApp (10.94s)
=== RUN   TestAccAblyKey
--- PASS: TestAccAblyKey (10.62s)
=== RUN   TestAccAblyNamespace
--- PASS: TestAccAblyNamespace (12.04s)
PASS
ok      terraform-provider-ably/internal/provider       33.897s
```

### CODE REVIEW

**Comment by graham-russell** [comment]
Changed this to draft until #51 is merged. Will rebase off of that so that the key resource can use v0.11.1 of the tf plugin framework.

**Review by kavalerov — APPROVED** [review]
LGTM, thanks!

P.S. Embedded capabilities also work well, so I think it is perfectly good enough for now.

File: docs/resources/api_key.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+
+
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Required
+
+- `app_id` (String) The Ably application ID which this key is associated with.
+- `capabilities` (Map of List of String) The capabilities that this key has. More information on capabilities can be found in the Ably documentation.
```
Comment (Morganamilo) [inline docs/resources/api_key.md:21]: Is it possible to Hyperlink here?

File: internal/provider/resource_ably_key.go [hunk 2]
```diff
@@ -0,0 +1,276 @@
+package ably_control
+
+import (
```
Comment (Morganamilo) [inline internal/provider/resource_ably_key.go:3]: I think these should re ordered to be std then external.

File: internal/provider/resource_ably_key.go [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+	// Gets the Ably App ID and Ably API Key ID value for the resource
+	app_id := state.AppID.Value
+	key_id := state.ID.Value
+
+	// Fetches all Ably Keys for the Ably App. The function invokes the Client Library Keys() method.
+	keys, _ := r.p.client.Keys(app_id)
+
+	// Loops through apps and if account id and key id match, sets state.
+	for _, v := range keys {
+		if v.AppID == app_id && v.ID == key_id {
```
Comment (Morganamilo) [inline internal/provider/resource_ably_key.go:156]: Should we check for non revoked also?

Comment (Morganamilo) [reply internal/provider/resource_ably_key.go:156]: Or rather match anyway but if it's revoked error out?

File: docs/resources/api_key.md (hunk 1 above)
Comment (graham-russell) [reply docs/resources/api_key.md]: Done

File: internal/provider/resource_ably_key.go (hunk 2 above)
Comment (graham-russell) [reply internal/provider/resource_ably_key.go:3]: Done

File: internal/provider/resource_ably_key.go (hunk 3 above)
Comment (graham-russell) [reply internal/provider/resource_ably_key.go]: Added check for revoked.

**Review by Morganamilo — APPROVED** [review]

**Pull request merged**